aws/resource_aws_dms_event_subscription.go

```diff
@@ -0,0 +1,251 @@
+				ForceNew: true,
+			},
+			"tags": {
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAwsDmsEventSubscriptionCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dmsconn
+
+	request := &dms.CreateEventSubscriptionInput{
+		Enabled:          aws.Bool(d.Get("enabled").(bool)),
+		SnsTopicArn:      aws.String(d.Get("sns_topic_arn").(string)),
+		SubscriptionName: aws.String(d.Get("name").(string)),
+		SourceType:       aws.String(d.Get("source_type").(string)),
+		Tags:             dmsTagsFromMap(d.Get("tags").(map[string]interface{})),
+	}
+
+	if v, ok := d.GetOk("event_categories"); ok {
+		request.EventCategories = expandStringList(v.(*schema.Set).List())
+	}
+
+	if v, ok := d.GetOk("source_ids"); ok {
+		request.SourceIds = expandStringList(v.(*schema.Set).List())
+	}
+
+	_, err := conn.CreateEventSubscription(request)
+
+	if err != nil {
+		return fmt.Errorf("Error creating DMS event subscription: %s", err)
+	}
+
+	d.SetId(d.Get("name").(string))
+
+	log.Println("[DEBUG] DMS create event subscription", request)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"creating", "modifying"},
+		Target:     []string{"active"},
+		Refresh:    resourceAwsDmsEventSubscriptionStateRefreshFunc(conn, d.Id()),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		MinTimeout: 10 * time.Second,
+		Delay:      10 * time.Second,
+	}
+
+	// Wait, catching any errors
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("error waiting for DMS event subscription (%s) creation: %s", d.Id(), err)
+	}
+
+	return resourceAwsDmsEventSubscriptionUpdate(d, meta)
+}
+
+func resourceAwsDmsEventSubscriptionUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dmsconn
+
+	request := &dms.ModifyEventSubscriptionInput{
+		Enabled:          aws.Bool(d.Get("enabled").(bool)),
+		SnsTopicArn:      aws.String(d.Get("sns_topic_arn").(string)),
+		SubscriptionName: aws.String(d.Get("name").(string)),
+		SourceType:       aws.String(d.Get("source_type").(string)),
+	}
+
+	if v, ok := d.GetOk("event_categories"); ok {
+		request.EventCategories = expandStringList(v.(*schema.Set).List())
+	}
+
+	log.Println("[DEBUG] DMS update event subscription:", request)
+
+	_, err := conn.ModifyEventSubscription(request)
+
+	if err != nil {
+		return fmt.Errorf("Error updating DMS event subscription: %s", err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"modifying"},
+		Target:     []string{"active"},
+		Refresh:    resourceAwsDmsEventSubscriptionStateRefreshFunc(conn, d.Id()),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		MinTimeout: 10 * time.Second,
+		Delay:      10 * time.Second, // Wait 30 secs before starting
+	}
+
+	// Wait, catching any errors
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("error waiting for DMS event subscription (%s) modification: %s", d.Id(), err)
+	}
+
```

Comment:
Tagging updates can be covered via:

```suggestion

	if d.HasChange("tags") {
		o, n := d.GetChange("tags")

		if err := keyvaluetags.DatabasemigrationserviceUpdateTags(conn, d.Get("arn").(string), o, n); err != nil {
			return fmt.Errorf("error updating DMS Event Subscription (%s) tags: %s", arn, err)
		}
	}
```